trunner/device.py

```diff
@@ -412,9 +466,10 @@ def load(self, test):
 
         phd = None
         load_dir = str(rootfs(test.target) / 'bin')
+        self.boot()
```

Comment:
As I mentioned earlier maybe `self.restart()` (or even `self.hard_restart()`) and keep self.boot as it was.

Reply:
ummm... why do we even need rebooting twice here?

Reply:
We need to turn Rpi ports on after powering up eval kit (because of partial supplying board through usb port, when it's powered off using relay - this causes some problems, so we avoid that). After starting the Rpi ports (it takes a while, because after restart there is pre-restart device on serial port, which has to disappear and after that there appears proper device) there is some output since restart, which disappears so we reset board second time to capture whole output.

Reply:
If there is an issue like that it needs to be written in the code in comment - othewise this looks like a bug/remnant of old code so I would delete it when working on the code :).

Regarding the issue:
 - the "old devices" issue may go away if You unbind the device directly via sysfs (see my other comment)
 - as I'm striving for as-fast-as possible reboots - we might try to circumvent it in some other way in the future (hmm... maybe cutting 5V from the cable would do it? Or maybe there is some other way of separating it or draining the voltage?)

Reply:
Ok, I will add some comment about that. I tried with cutting 5V and with desoldering a zero resistor, which connect board 5V_SYS and 5V_USB_OTG. From what I remember then psu fails (port wasn't available)